migrations/20241120004033-create-calculate-last-day-usage-procedure.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.query(`
+      CREATE OR REPLACE PROCEDURE calculate_last_day_usage()
+        LANGUAGE plpgsql
+        AS $$
+        BEGIN
+        INSERT INTO public.usages (id, user_id, delta, period, type, created_at, updated_at)
+        SELECT
+            uuid_generate_v4() AS id,
+            u.uuid::uuid AS user_id,
+            SUM(
+                CASE 
+                    WHEN f.status = 'DELETED' AND date_trunc('day', f.created_at) = date_trunc('day', f.updated_at) THEN 0
+                    WHEN f.status = 'DELETED' THEN -f.size
+                    ELSE f.size
+                END
+            ) AS delta,
+            CURRENT_DATE - INTERVAL '1 day' AS period,
+            'monthly' AS type,
+            NOW() AS created_at,
+            NOW() AS updated_at
+        FROM
+            files f
+        JOIN users u ON u.id = f.user_id
+        LEFT JOIN (
+            SELECT user_id, MAX(period) AS last_period
+            FROM public.usages
+            GROUP BY user_id
+        ) mru
```

Comment:
If a user does not have a record in usage, we would like not to calculate their usage so we can reduce usecases such as an user having their last day usage calculated but files created before this implemented being ignored.

Reply:
I do not get that. If we create the first row on demand for already existing users, which could be the issue?